swarm/api/inspector.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2016 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package api
+
+import (
+	"context"
+
+	"github.com/ethereum/go-ethereum/swarm/network"
+	"github.com/ethereum/go-ethereum/swarm/storage"
+)
+
+type Inspector struct {
+	api      *API
+	hive     *network.Hive
+	netStore *storage.NetStore
```

Comment:
Why Is this a `NetStore` and not a `LocalStore`? I'd expect for this to be `LocalStore` as we care only about the local node, and don't care about any retrieve requests, etc. `NetStore` in general first checks the local node, and if a chunk is not present, makes a request with a fetcher.

Reply:
Because the main `Swarm` object, which creates this API, doesn't store the `LocalStore` object, only `NetStore`.

Also, as this is a general Inspection API, it may be more useful later if we need some more inspection tools, and `NetStore` is broader.

Reply:
Ok, fair enough.